### PR TITLE
Add an option to authenticate using `grant_type=client_credentials`

### PR DESCRIPTION
**Pull request by pkaczynski**
There is no option to authenticate using `clien_credentials` (just `client_id` and `client_secret`).

### CODE REVIEW

File: src/sasctl/core.py [hunk 1]
```diff
@@ -712,7 +741,14 @@ def get_auth(self, username=None, password=None, token=None):
 
         # If explicit username & password were provided, use them.
         if username is not None and password is not None:
-            return self.get_oauth_token(username, password)
+            return self.get_oauth_token(
+                username, password
+            )
+        elif client_id is not None and client_secret is not None:
```
Comment (smlindauer) [inline src/sasctl/core.py:747]: This additional elif with a return embedded is throwing a [PYL-R1705 linter error](https://deepsource.io/gh/sassoftware/python-sasctl/run/249aba1b-cecd-45fa-8ba9-e8acd8dfa4d5/python/PYL-R1705?listindex=0). For the time being can we split this out to two separate if statements.

In the future, it looks like this function could use a refactor for two elements:
1. Combine if statements into fully encapsulated set of cases (similar to the switch/case implementation in Python 3.10).
```
if username and password:
    return ...
elif client_id and client_secret:
    return ...
elif token:
    return ...
else:
    try:
        return kerberos
    ...
    prompt user input
    return user input
```
2. Remove double negative conditionals.
`if username is not None and password is not None:` can be written as `if username and password:`

File: src/sasctl/core.py [hunk 2]
```diff
@@ -745,22 +781,27 @@ def get_oauth_token(
         client_id=None,
         client_secret=None,
     ):
-        """Request an OAuth2 access token using either a username & password or an auth token.
+        """Request an OAuth2 access token using either a username & password,
```
Comment (smlindauer) [inline src/sasctl/core.py:784]: Oxford comma please:
"[...] username & password, client_id [&] client_secret, or an auth token.

**Review by smlindauer — COMMENTED** [review]
Everything not commented on looks good to me. 

The deepsource.io check is throwing another error on [line 557](https://github.com/sassoftware/python-sasctl/blob/2fb353959a5c3f21d53cd3e787048ed27e9de11b/src/sasctl/core.py#L557-L557), but wasn't previously throwing an error for it (or it had been ignored). @jlwalke2: Do you think this is worth a refactor or should we just skip the error in deepsource.io?

File: src/sasctl/core.py (hunk 1 above)
Comment (pkaczynski) [reply src/sasctl/core.py]: Don, point 1.

**Pull request merged**